engine/src/main/java/com/redhat/cloud/notifications/processors/camel/CamelTypeProcessor.java

```diff
@@ -133,7 +134,11 @@ private JsonObject buildPayload(Event event, Endpoint endpoint, String originalE
 
         if (featureFlipper.isSourcesUsedAsSecretsBackend()) {
             // Get the basic authentication and secret token secrets from Sources.
-            secretUtils.loadSecretsForEndpoint(endpoint);
+            try {
+                this.secretUtils.loadSecretsForEndpoint(endpoint);
+            } catch (final SourcesException e) {
+                Log.error(e.getMessage(), e);
+            }
```

Comment:
I forgot to comment this case.

We're processing a Camel endpoint and the secret cannot be loaded from Sources for some reason. I think the exception thrown here should bubble up to the `EventConsumer` try/catch block. It does really make sense to continue processing the endpoint if the secret is missing.

Reply:
Agree - and maybe have some sort of retry mechanism

Reply:
Done!